docs/xgc.md

```diff
@@ -116,6 +131,12 @@ Options that change the behavior of the Garbage Collector (GC).
 
     The default value is 95, which means that anything over 5% of total application run time spent on GC is deemed excessive. This option can be used only when [`-Xenableexcessivegc`](xenableexcessivegc.md) is set (enabled by default).
 
+### `HierarchicalScanOrdering`
+
+        -Xgc:hierarchicalScanOrdering
+
+: This option sets the scan mode for the scavenge operation (`gencon` GC policy) to hierarchical mode. This mode reflects the method for traversing the object graph and adds partial depth first traversal on top of breadth first scan mode. The aim of hierarchical mode is to minimize object distances. This option is the default for the `gencon` GC policy.
+
```

Comment:
There is implementation for Balanced as well. @amicic Should we mention it here too?

Reply:
No, Balanced GC does not have hierarchical. The option only applies to Gencon.
So, Balanced has a choice of two and Gencon a choice of 3.

(BTW, any of these options do _not_ apply to any of non-generational GC policies, like optthruput).